client/extensions/woocommerce/app/order/order-customer/dialog.js

```diff
@@ -0,0 +1,205 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { get, noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import AddressView from 'woocommerce/components/address-view';
+import Button from 'components/button';
+import Dialog from 'components/dialog';
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormLegend from 'components/forms/form-legend';
+import FormPhoneMediaInput from 'components/forms/form-phone-media-input';
+import FormTextInput from 'components/forms/form-text-input';
+import getAddressViewFormat from 'woocommerce/lib/get-address-view-format';
+
+// @todo Update this to use our store countries list
+import countriesListBuilder from 'lib/countries-list';
+const countriesList = countriesListBuilder.forPayments();
+
+class CustomerAddressDialog extends Component {
+	static propTypes = {
+		address: PropTypes.shape( {
+			address_1: PropTypes.string.isRequired,
+			address_2: PropTypes.string,
+			city: PropTypes.string.isRequired,
+			state: PropTypes.string,
+			country: PropTypes.string.isRequired,
+			postcode: PropTypes.string,
+			email: PropTypes.string,
+			first_name: PropTypes.string.isRequired,
+			last_name: PropTypes.string.isRequired,
+			phone: PropTypes.string,
+		} ),
+		closeDialog: PropTypes.func,
+		isBilling: PropTypes.bool,
+		isVisible: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		address: {
+			street: '',
+			street2: '',
+			city: '',
+			state: 'AL',
+			country: 'US',
+			postcode: '',
+			email: '',
+			first_name: '',
+			last_name: '',
+			phone: '',
+		},
+		closeDialog: noop,
+		isBilling: false,
+		isVisible: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			address: props.address,
+			phoneCountry: 'US',
```

Comment:
Might I suggest here (and above, elsewhere) that the default country be informed by the merchant's country.  It would be better for Canadian merchants, for example, to see Canada as the default so they don't have to keep changing it.

You might even want to go ahead and use the merchant's state/province to pre-select that too.